worker/src/RTC/IceServer.cpp

```diff
@@ -734,4 +853,88 @@ namespace RTC
 		// Notify the listener.
 		this->listener->OnIceServerSelectedTuple(this, this->selectedTuple);
 	}
+
+	void IceServer::StartConsentCheck()
+	{
+		MS_TRACE();
+
+		MS_ASSERT(IsConsentCheckSupported(), "ICE consent check not supported");
+		MS_ASSERT(!IsConsentCheckRunning(), "ICE consent check already running");
+		MS_ASSERT(this->selectedTuple, "no selected tuple");
+
+		// Create the ICE consent check timer if it doesn't exist.
+		if (!this->consentCheckTimer)
+		{
+			this->consentCheckTimer = new TimerHandle(this);
+		}
+
+		this->consentCheckTimer->Start(this->consentTimeoutMs);
+	}
+
+	void IceServer::RestartConsentCheck()
+	{
+		MS_TRACE();
+
+		MS_ASSERT(IsConsentCheckSupported(), "ICE consent check not supported");
+		MS_ASSERT(IsConsentCheckRunning(), "ICE consent check not running");
+		MS_ASSERT(this->selectedTuple, "no selected tuple");
+
+		this->consentCheckTimer->Restart();
+	}
+
+	void IceServer::StopConsentCheck()
+	{
+		MS_TRACE();
+
+		MS_ASSERT(IsConsentCheckSupported(), "ICE consent check not supported");
+		MS_ASSERT(IsConsentCheckRunning(), "ICE consent check not running");
```

Comment:
This would suffice.

In general, if we the check is running, we can assume it is supported. And we it's enough to check for support only when start running.

```suggestion
		MS_ASSERT(IsConsentCheckRunning(), "ICE consent check not running");
```

Reply:
same